qt/applications/workbench/workbench/plotting/figureinteraction.py

```diff
@@ -55,13 +58,22 @@ def __init__(self, fig_manager):
 
         canvas = fig_manager.canvas
         self._cids = []
-        self._cids.append(canvas.mpl_connect('button_press_event',
-                                             self.on_mouse_button_press))
+        self._cids.append(canvas.mpl_connect('button_press_event', self.on_mouse_button_press))
+        self._cids.append(canvas.mpl_connect('button_release_event', self.on_mouse_button_release))
+        self._cids.append(canvas.mpl_connect('draw_event', self.draw_callback))
+        self._cids.append(canvas.mpl_connect('motion_notify_event', self.motion_event))
+        self._cids.append(canvas.mpl_connect('resize_event', self.mpl_redraw_annotations))
 
         self.canvas = canvas
         self.toolbar_manager = ToolbarStateManager(self.canvas.toolbar)
+        self.toolbar_manager.home_button_connect(self.redraw_annotations)
         self.fit_browser = fig_manager.fit_browser
         self.errors_manager = FigureErrorsManager(self.canvas)
+        self.markers = []
+        self.vertical_markers = []
+        self.valid_lines = [str(name) for name in ['solid', 'dashed', 'dotted', 'dashdot']]
+        self.valid_colors = {'blue': 'C0', 'orange': 'C1', 'green': 'C2', 'red': 'C3', 'purple': 'C4'}
```

Comment:
Just as a code style thing could we define these as constants at the top of the file.